src/App.js

```diff
@@ -1,22 +1,45 @@
-import React from "react";
-import { BrowserRouter, Route, Switch } from "react-router-dom";
+import React, { useState } from "react";
+import { BrowserRouter, Route, Switch, Redirect } from "react-router-dom";
 import "./App.css";
 import NavBar from "./Components/NavBar/NavBar";
 import Auth from "./Components/Auth/auth";
 import Bookings from "./Components/Bookings/Bookings";
 import Trips from "./Components/Trips/Trips";
+import AuthContext from "./Components/Context/context";
 
 function App() {
+	const [token, setToken] = useState(null);
+	const [userId, setUserId] = useState(null);
+
+	const login = (token, userId, tokenExpiration) => {
+		setToken(token);
+		setUserId(userId);
+	};
+
+	const logout = () => {
+		setToken(null);
+		setUserId(null);
+	};
 	return (
 		<BrowserRouter>
-			<NavBar />
-
 			<React.Fragment>
-				<Switch>
-					<Route path="/auth" component={Auth} />
-					<Route path="/bookings" component={Bookings} />
-					<Route path="/trips" component={Trips} />
-				</Switch>
+				<AuthContext.Provider
+					value={{
+						token: token,
+						userId: userId,
+						login: login,
+						logout: logout,
+					}}>
+					<NavBar onLogout={logout} />
+					<Switch>
+						<Route path="/auth" component={Auth} />
```

Comment:
Create a custom protectedRoutes for all routes that requires user to be authenticated 

Something like this

```js

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route {...rest} render={(props) => (
     token
      ? <Component {...props} />
      : <Redirect to={{
          pathname: '/auth',
          state: { from: props.location }
        }} />
  )} />
)

// Then replace the routes with 
<ProtecedRoute path="/bookings" component ... />
```

Reply:
You can read more about this concept here(Don't watch video. Just read) => https://tylermcginnis.com/react-router-protected-routes-authentication/